Description: Xiaomi QR code is parsed incorrectly.
 Xiaomi use own template.
 .
 wifi-qr (0.2-3) unstable; urgency=medium
 .
   * #18 Xiaomi QR code is parsed incorrectly.
   * #17 QR issue when the name and SSID differ.
   * #16 QR issue when the password has special characters.
   * #15 QR issue when the SSID has special characters.
   * #12 Password with special characters needs to be unquoted.
Author: kokoye2007 <kokoye2007@gmail.com>

---
Origin: upstream, https://github.com/kokoye2007/wifi-qr
Forwarded: not-needed
Reviewed-By: kokoye2007 <kokoye2007@gmail.com>
Last-Update: 2023-05-21

--- wifi-qr-0.2.orig/wifi-qr
+++ wifi-qr-0.2/wifi-qr
@@ -258,16 +258,16 @@ call_wifi_scan() {
 	WIFIQRDATA="${wifiqrdata}"
 
 	# Extract the SSID
-	QSSID=$(echo "$WIFIQRDATA" | sed -n 's/.*S:\(.*\);T.*/\1/p' |  sed 's/\\\([\\;,:"\]\)/\1/g' )
+	QSSID=$(echo "$WIFIQRDATA" | sed -n 's/.*S:\([^;]*\)\(;[PHT]\).*$/\1/p' |  sed 's/\\\([\\;,:"\]\)/\1/g' )
 
 	# Extract the security type
-	QTYPE=$(echo "$WIFIQRDATA" | sed -n 's/.*T:\(.*\);P.*/\1/p')
+	QTYPE=$(echo "$WIFIQRDATA" | sed -n 's/.*T:\([^;]*\)\(;[PHS]\).*$/\1/p' |  sed 's/\\\([\\;,:"\]\)/\1/g' )
 
 	# Extract the password
 	if [[ $WIFIQRDATA == *';H:'* ]]; then
-	  QKEES=$(echo "$WIFIQRDATA" | sed -n 's/.*;P:\(.*\);H:.*/\1/p' | sed 's/\\:/:/g' | sed 's/\\\([\\;,:"\]\)/\1/g'  )
+	  QKEES=$(echo "$WIFIQRDATA" | sed -n 's/.*P:\([^;]*\)\(;[TSH]\).*$/\1/p' |  sed 's/\\\([\\;,:"\]\)/\1/g' )
 	else
-	  QKEES=$(echo "$WIFIQRDATA" | sed -n 's/.*;P:\(.*\);;.*/\1/p' | sed 's/\\:/:/g' | sed 's/\\\([\\;,:"\]\)/\1/g' )
+	  QKEES=$(echo "$WIFIQRDATA" | sed -n 's/.*P:\([^;]*\)\(;[TS]\).*$/\1/p' |  sed 's/\\\([\\;,:"\]\)/\1/g' )
 	fi
 
 	# Check if the Wi-Fi network is hidden
